src/types/params.ts

```diff
@@ -12,8 +12,16 @@ export type ParamGetSet<T = any> = {
 
 export type Param = ParamGetter | ParamGetSet | RegExp | BooleanConstructor | NumberConstructor | StringConstructor
 
+function isNotConstructor(value: Param): boolean {
+  return value !== String && value !== Boolean && value !== Number
+}
+
 export function isParamGetter(value: Param): value is ParamGetter {
-  return typeof value === 'function' && value !== Boolean && value !== Number
+  return typeof value === 'function' && isNotConstructor(value)
+}
+
+export function isParamSetter(value: Param): value is ParamSetter {
+  return typeof value === 'function' && isNotConstructor(value)
```

Comment:
What's your thought here?

Reply:
there might not be much utility in defining a ParamSetter type at all, so this function likely goes away? Let's see where we land in the other thread

Reply:
It only exists for ParamGetSet. 

But I'm mostly wondering about the isNotConstructor piece. Those technically are valid. From a types perspective. You just thinking they're different from a runtime perspective because we check for and override them?

Reply:
ah, I see. I only pulled that out to not repeat myself in `IsParamSetter`, the logic inside was already defined in `IsParamGetter`. Though I did add "value !== String" since we now have a string param we use if we are given `String` for a param.